### ISSUE TITLE
How its works ?

### ISSUE THREAD

**Issue by Thibaut-ids** [ansible-semaphore/semaphore#226]
Hello,
I'm new on github and semaphore. I do not quite understand how it works. I don't find documentation with a simple example. Can someone give me a simple example of creating a project from the beginning until its execution? Thanks a lot.

**Comment by mikequentel** [comment]
Hi @Thibaut-ids I am trying to figure out the same information. From https://code-complete.com/code/?p=40 it looks like one does the following (_just my guesses at this point_):

1. **Add Playbook** specify:
  * **Name**: <arbitrary value>
  * **Playbook Location**: <git URL to playbook>
  * **Identity**: <ssh key data that was put into Key Store>

1. Based on the above, I have done the following:
  1. Login to Semaphore
  1. Input SSH key into **Keystore** -- it requires input of both public and private keys.
  1. Specify **Name** and **Git URL** for **Playbook Repsositories**
  1. Specify **Inventory** as the name of the hosts inventory file (eg: `hosts`).
  1. In **Task Templates** create a new template specifying the following:
    * **Playbook Name**: name of the `*.yml` file to run.
    * **SSH Key**: the SSH key identity specified earlier.
    * **Inventory**: the inventory file specified earlier.
    * **Playbook Repository**: the git URL specified earlier.

**Comment by Thibaut-ids** [comment]
Hi mikequentel,
Sorry for my response time ... I'm no longer in my enterprise at the moment. I will try it in one or two weeks. I'll keep you informed.
Anyway thanks for taking the time to make a constructed answer.
That seems a lot clearer now.

ps: sorry for my english ...

**Comment by mikequentel** [comment]
De rien...j'espère qu'il est utile. :smile:

**Comment by deepikapasupulati** [comment]
how to know the web url of semaphore after semaphore -setup??
Kindly help in the installation 
Thanks in advance

**Comment by mikequentel** [comment]
Hi @deepikapasupulati try http://yourhostname:3000/auth/login --the default port is `3000` 

After running `semaphore -setup` then launch the server with the `--config` flag specifying the location of the generated `semaphore_config.json` file (generated from `-setup`).

I launch the server using a shell script named `start.sh` which contains this content:
```
nohup /opt/semaphore/semaphore -config /home/semaphore/semaphore_config.json >> semaphore.log 2>&1 &
```

**Comment by deepikapasupulati** [comment]
Thank you so much I did the same way as you suggested I launched the server using  
 semaphore -config ./semaphore -config /home/semaphore/semaphore_config.json

Later I tried to open  http://yourhostname:3000/auth/login but it showing me loading but nothing was happening. 
**semaphore Dashboard
 
Users
 
 {{ user.name }}
 
  
semaphore v{{ semaphore.version }} 
 System info 

 Log out 

Loading...**

And in the CLI it is showing 
[GIN] 2016/11/28 - 11:07:37 | 200 |      352.48µs | 10.203.192.116 |   GET     /auth/login
[GIN] 2016/11/28 - 11:07:37 | 200 |   28.915231ms | 10.203.192.116 |   GET     /public/css/semaphore.css
[GIN] 2016/11/28 - 11:07:37 | 200 |   18.058151ms | 10.203.192.116 |   GET     /public/vendor/fontawesome/fonts/fontawesome-webfont.eot
[GIN] 2016/11/28 - 11:07:37 | 200 |   78.735358ms | 10.203.192.116 |   GET     /public/js/bundle.js

Please help me out.

**Comment by matejkramny** [comment]
@deepikapasupulati looks like the front end JS is not loading or there is some sort of an error.

It is also strange that serving up static files takes 80ms. Can you describe the architecture of your system?

**Comment by deepikapasupulati** [comment]
Presently working on RHEL 7 server. I am not sure whether semaphore is listening to port 3000 or not. I even tied with sudo firewall-cmd --zone=public --add-port=3000/tcp --permanent.

**Comment by matejkramny** [comment]
does it respond to port 3000 with curl when logged into the server?

Could it be SELinux or similar preventing it from serving up content?

**Comment by deepikapasupulati** [comment]
curl: (7) Failed to connect to 127.0.0.1 port 3000: Connection refused
There is no SElinux installed.

**Comment by matejkramny** [comment]
looks like the process isn't running or listening to 3000. Maybe it can't bind to the port?

**Comment by deepikapasupulati** [comment]
Is there is error log which I can review the error?

**Comment by matejkramny** [comment]
It should say in the output although I am not sure it will say if it fails to listen to a port

**Comment by deepikapasupulati** [comment]
When I do semaphore -setup

Generated configuration:
 {
        "mysql": {
                "host": "127.0.0.1:3306",
                "user": "root",
                "pass": "cloud01",
                "name": "semaphore"
        },
        "port": "",
        "bugsnag_key": "",
        "tmp_path": "/home/semaphore",
        "cookie_hash": "kL3/1lHJQHhg0fo7hxW1fqb4NFOhUpx1wPhGPriFWAE=",
        "cookie_encryption": "Uz+QK30vfkNqH9f2pMpm7KZn/rCHEqTWWioaY82Niic="
 }
Why it is not taking port 3000 as default?

**Comment by matejkramny** [comment]
It does: https://github.com/ansible-semaphore/semaphore/blob/master/util/config.go#L116

You can try making it explicit, such as `"port": ":3000"`.

**Comment by deepikapasupulati** [comment]
cat semaphore_config.json
{
        "mysql": {
                "host": "127.0.0.1:3306",
                "user": "root",
                "pass": "cloud01",
                "name": "semaphore"
        },
        "port": "",
        "bugsnag_key": "",
        "tmp_path": "/dst/home/cloud",
        "cookie_hash": "AL11Adm2bUsQAUjZUd/ECfCBUhZ8Ogi4Tpyyp7Do/nE=",
        "cookie_encryption": "2v8vztjtyGzwqTDdbJOziUScSpq9YALG5Gfwckyxdcc="
 }
[root@devl21425005 cloud]# semaphore -printConfig
{
        "mysql": {
                "host": "127.0.0.1:3306",
                "user": "root",
                "pass": "",
                "name": "semaphore"
        },
        "port": ":3000",
        "bugsnag_key": "",
        "tmp_path": "/tmp/semaphore",
        "cookie_hash": "L1dLQTX7tcgqgTbCxsF3O6/X82UInR1e1HuTCnJlPU4=",
        "cookie_encryption": "FItYBzCT4twbFiBsy78zjQPHRwiIs00UUt8tB74EUks="
}

Why these two outputs are different?
In config file the port number is null.

**Comment by matejkramny** [comment]
`printConfig` flag prints the default configuration

**Comment by deepikapasupulati** [comment]
Even though I installed MySQL 
I am getting this error
 #semaphore
Semaphore 2.0.4
Port :3000
MySQL root@127.0.0.1:3306 semaphore
Tmp Path (projects home) /tmp/semaphore
[root@devl21425005 /]# panic: Error 1045: Access denied for user 'root'@'localhost' (using password: NO)

goroutine 1 [running]:
panic(0xa6f0c0, 0xc8201f2120)
        /usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:464 +0x3e6
main.main()
        /Users/m/p/Projects/semaphore/cli/main.go:43 +0x5a7
2016/11/28 15:30:50 notifying bugsnag: Error 1045: Access denied for user 'root'@'localhost' (using password: NO)
2016/11/28 15:30:50 bugsnag.Notify: not notifying in development

**Comment by deepikapasupulati** [comment]
When I do semaphore -setup
> Is this correct? (yes/no): yes
 Running: mkdir -p /home/semaphore..
 Configuration written to /home/semaphore/semaphore_config.json..
 Pinging database..

 Running DB Migrations..
Skipping v0.0.0
Skipping v1.0.0
Skipping v1.1.0
Skipping v1.2.0
Skipping v1.3.0
Skipping v1.4.0
Skipping v1.5.0
Skipping v0.1.0
Skipping v1.6.0
Skipping v1.7.0


It is skipping the DB Migrations

**Comment by matejkramny** [comment]
That's because they've already been executed, unless it's a new/blank database

**Comment by matejkramny** [comment]
What's the mysql error about?

**Comment by deepikapasupulati** [comment]
Error 1045: Access denied for user 'root'@'localhost' (using password: NO) 
What is this error about?

**Comment by deepikapasupulati** [comment]
When we setup semaphore  Running DB Migrations..
Creating migrations table
Executing migration v0.0.0 (at 2016-06-01 10:15:14.375431485 +0300 EEST)...
 [11/11]
Executing migration v1.0.0 (at 2016-06-01 10:15:14.796656259 +0300 EEST)...
 [7/7]
Executing migration v1.1.0 (at 2016-06-01 10:15:15.195906597 +0300 EEST)...
 [1/1]
Executing migration v1.2.0 (at 2016-06-01 10:15:15.260100655 +0300 EEST)...
 [1/1]
Executing migration v1.3.0 (at 2016-06-01 10:15:15.291314552 +0300 EEST)...
 [3/3]
Executing migration v1.4.0 (at 2016-06-01 10:15:15.818140197 +0300 EEST)...
 [2/2]
Executing migration v1.5.0 (at 2016-06-01 10:15:15.917563871 +0300 EEST)...
 [1/1]
Executing migration v0.1.0 (at 2016-06-01 10:15:15.947447735 +0300 EEST)...
 [6/6]

It should be like this right?
But when I do it is skipping. Don't know why.

**Comment by matejkramny** [comment]
Yeah that's right. It only runs each migration once. If you look inside the database, there's a migrations table with entry of each migration version.

That mysql error looks like the user has a password, but none provided.

Example: `mysql -uroot` when root user has a password set, produces `(using password: NO)` whereas providing the wrong password produces a different error (`using password: YES`)

**Comment by deepikapasupulati** [comment]
Tried in other server but facing with the same error
**semaphore Dashboard

Users

{{ user.name }}

semaphore v{{ semaphore.version }}
 System info

Log out

Loading...**
[GIN-debug] Listening and serving HTTP on :3000
[GIN] 2016/11/29 - 11:17:47 | 200 |     410.755µs | 10.203.192.116 |   GET     /auth/login
[GIN] 2016/11/29 - 11:18:44 | 200 |     219.121µs | 10.203.192.116 |   GET     /auth/login
[GIN] 2016/11/29 - 11:18:44 | 200 |   33.933437ms | 10.203.192.116 |   GET     /public/css/semaphore.css
[GIN] 2016/11/29 - 11:18:44 | 200 |    7.138865ms | 10.203.192.116 |   GET     /public/vendor/fontawesome/fonts/fontawesome-webfont.eot
[GIN] 2016/11/29 - 11:18:44 | 200 |   56.155533ms | 10.203.192.116 |   GET     /public/js/bundle.js
[GIN] 2016/11/29 - 11:18:46 | 200 |     204.308µs | 10.203.192.116 |   GET     /auth/login
[GIN] 2016/11/29 - 11:18:46 | 200 |   15.461254ms | 10.203.192.116 |   GET     /public/css/semaphore.css
[GIN] 2016/11/29 - 11:18:46 | 200 |    2.184471ms | 10.203.192.116 |   GET     /public/vendor/fontawesome/fonts/fontawesome-webfont.eot
[GIN] 2016/11/29 - 11:18:46 | 200 |   56.346262ms | 10.203.192.116 |   GET     /public/js/bundle.js


Please help me out.
Thank you in advance.

**Comment by deepikapasupulati** [comment]
What do you mean by this
# panic: open nohup: no such file or directory
When I tried to launch the program I got this error

**Comment by matejkramny** [comment]
Can you paste all of the stack trace?

**Comment by deepikapasupulati** [comment]
Thanks for your help. It worked on my local server but it is not working in my enterprise(office) servers. As I am using the proxy server to download the semaphore link. Does it requires the complete internet access to run semaphore?

**Comment by matejkramny** [comment]
It shouldn't need to have internet connection on the server.

**Comment by deepikapasupulati** [comment]
Then when I tried in my local sever which has internet access it worked. When  tried in office server it is showing **semaphore Dashboard

Users

{{ user.name }}

semaphore v{{ semaphore.version }}
 System info

Log out

Loading...**
 never got the complete loaded page. 
Please help me.

**Comment by deepikapasupulati** [comment]
Is it mandatory to install ansible on semaphore machine?

**Comment by matejkramny** [comment]
Yes @deepikapasupulati 

Dependencies: https://github.com/ansible-semaphore/semaphore/wiki/Installation#dependencies

**Comment by deepikapasupulati** [comment]
Thank you so much it helped a lot.

**Comment by deepikapasupulati** [comment]
Hey hi,

When I use my office public repository to clone the .yml file, while running the playbook it is showing as 
11:57:25: access key TEST installed
11:57:25: Cloning repository
11:57:25: Cloning into 'repository_5'...
11:57:25: Host key verification failed.
11:57:25: fatal: Could not read from remote repository.
11:57:25: 
11:57:25: Please make sure you have the correct access rights
11:57:25: and the repository exists

Though I have access to that repo.
And one more thing in key store what key should be placed? (host server key or repository key)
Please help me.
Thanks in advance!!

**Comment by matejkramny** [comment]
May be something with the ssh agent, but to resolve, ssh into the targets manually first to accept it's RSA fingerprint.

**Comment by paulinomreyes** [comment]
Hello! I am new to Semaphore and trying to learn Ansible. I managed to deploy Semaphore and although I was able to create and successfully run templates/playbooks, I am not too sure about the use of Environment. All I did to successfully create a Task Template was to give it a dummy Environment whose content is just {}. Can anyone provide more info on what's the use Environment?

**Comment by matejkramny** [comment]
@paulinomreyes with ansible you can specify a json file as input. This is what semaphore provides to ansible.

**Comment by paulinomreyes** [comment]
Thanks @matejkramny . I am trying to create a How-To document for this similar to @mikequentel replied above. Guess I need to really learn Ansible first. Btw, is Github the only repository (for playbook files) supported for Semaphore?

**Comment by mikequentel** [comment]
Hi @paulinomreyes please see the note on **Playbook Repository** above--can use other Git URLs, in addition to GitHub. I have successfully run semaphore using repos stored in GitLab.

**Comment by Koxx3** [comment]
I don't understand how it works...

I specified "hosts" and inventory name, and for content, I just placed 
`windows`
(I am trying to win_ping some windows ... to start).

In Environnement, I placed only 
`{
windows
}
`

The content of my hosts file is : 
`
[windows]
192.168.33.10
192.168.33.11
`

and my test.yml is : 
`---
# This playbook uses the win_ping module to test connectivity to Windows hosts
- name: Ping windows
  hosts: windows

  tasks:
  - name: ping windows
    win_ping:
`

Result : 
it is trying to connect in SSH to a server called 'windows' ...
I al clearly mossing something ...

The following command works fine : 
`ansible-playbook test.yml -vvvv -i hosts`

Any help would be welcome :(

My project:
https://github.com/Koxx3/ansible-semaphore

**Comment by Koxx3** [comment]
arghh...

reading the issues, I finally found the inventory and some environnement content samples ... and it works ! 

inventory name : hosts
intentory content : 
`[windows]
192.168.33.11`

environnement content : 
`{
hosts=windows
}`

I clear tutorial would be awesome. I lost 3 hours trying to get semaphore works :(

**Comment by dropped** [comment]
Hi @Koxx3, could you provide a playbook's sample that works?
I lost some hours as well, I don't know if my directory structure is wrong, if is missing something.

**Comment by kranjith490** [comment]
hey guys,am facing a problem while running the command semaphore -setup 

 Running DB Migrations..
Skipping v0.0.0
Skipping v1.0.0
Skipping v1.1.0
Skipping v1.2.0
Skipping v1.3.0
Skipping v1.4.0
Skipping v1.5.0
Executing migration v0.1.0 (at 2017-10-24 03:16:57.697281688 -0400 EDT)...
 [1/6]
 ERR! Query: alter table user change `created` `created` datetime not null default current_timestamp

Rolling back v0.1.0 (time: 2017-10-24 03:16:57.697767401 -0400 EDT)...
Rollback SQL doesn't exist.


 Database migrations failed!
 Error 1067: Invalid default value for 'created'
I dono why it is throwing this kind of error kindly help me out
@deepikapasupulati (you too faced the issue I think)
@mikequentel kindly help me

**Comment by mikequentel** [comment]
Hi @kranjith490 the author of the [blog](https://code-complete.com/code/?p=40) says one should see the instructions at: https://github.com/ansible-semaphore/semaphore/wiki/Installation

**Comment by gowtamk** [comment]
First of all i would like to say this tool very well written.
I was trying to compile this in my setup, i tried to get all the dependencies cloned from different repositories. Except for one package  that is MuleKick.   
Mulekick is not found anywhere in the web.  I need help here,  first of all where can i find Mulekick, if it is not available what is the alternative to mulekick package and how do I use it?

**Comment by matejkramny** [comment]
Hi @gowtamk, this issue will be soon fixed once #585 is merged

**Comment by danielo515** [comment]
So, I have the same question. How the hell does this work? When I execute task, despite the inventori is correct and parsed it allways executes it against localhost.
How is this supposed to work?
